### ISSUE TITLE
Minor: 0 in a conditional is somewhat inconsistent. 

### ISSUE THREAD

**Issue by cycomachead** [jmoenig/Snap--Build-Your-Own-Blocks#871]
I don't know that it's worth fixing this one anytime soon, since it could have possible reprocussions, but this case is a bit odd for a Snap<i>!</i> user. (Though it's fairly clear what's going in JS.)

![blocks](https://cloud.githubusercontent.com/assets/1505907/8696307/062595ce-2aa0-11e5-9daf-8676f4f30008.png)

I've honestly seen TA's run into this more than students have, because students are usually better about trying to use explicitly T/F values. (At least if they are listening to us...), but overall this is a minor issue.

**Comment by jmoenig** [comment]
yeah, we could (and maybe at one time should) type check every primitive's input. This is going to (slightly) slow down performance, but types seem to be a consistently strong topic here (personally I think they're totally overrated).

**Comment by cycomachead** [comment]
This one is small enough that it's not such a huge deal, but it is hard to explain the difference between a string that is a number and the number literal.

**Comment by jmoenig** [comment]
well the point is that technically they're *both* bugs. We've had this before, right, what if something *does* work because of some quirk in JS but isn't *guaranteed* to work in Snap....

**Comment by cycomachead** [comment]
Yeah, I mean there's no real way of knowing unfortunately. :/ I guess, I lean towards fixing it for future projects, though it would be nice to have a better way of listing such changes if users run into problems.

My gut would be that more "programmers" expect 0 to be false, but they're definitely not the target audience. I'm honestly not too sure what the intuition for a new user would be. "0 is false" almost seems like a "bug" if every other number is truthy...hmm. I was going to suggest mirroring BYOB, but BYOB makes it such that only explicitly true values are truthy, which just seems wrong to me...

Maybe Brian would have some thoughts about this.

**Comment by jmoenig** [comment]
I think this whole "truthy" and "falsey" thing is wrong, and the *right* thing would be to only accept `false` as false, and everything else as true. Alternatively raise an error if the input isn't of type Boolean.

**Comment by bb010g** [comment]
:+1: for only `<false>` being falsy.

**Comment by cycomachead** [comment]
> I think this whole "truthy" and "falsey" thing is wrong, and the right thing would be to only accept false as false, and everything else as true

There's a small chance this could break a few existing projects, but I (and a few TAs) agree that this is also a good way to go.

**Comment by brianharvey** [comment]
I agree that only False should be falsey.  I am of two minds as to whether everything else should be truthy (as in God's language) or only True should be truthy (as in Logo).  The advantage of the former is that you can have semipredicates like MEMBER (returns a list or #f) which prevents having to traverse a list twice.  The advantage of the latter is, of course, that it makes sense to the novice.

**Comment by cycomachead** [comment]
> (Edit: Another advantage of everything but False being truthy is that it maintains the Scratch ideal of avoiding error messages.)

Interestingly, Scratch 2 doesn't let you drop anything other than a boolean into a conditional, so they can guarantee only T/F values. I think only false being false is reasonable to students. No matter the path there is always a question of "what about the wrong input" and treating everything else as true is at least consistent.
